batch/parsers/xUnitParser.js

```diff
@@ -24,55 +24,57 @@ class XUnitParser {
 			name: testCaseXMLObject.name,
 			status: 'ok',
 			timestamp: `${parsedData.suite.timestamp}`,
-			location:`${GITHUB_REPO_URL}${this.configuration.repo}${this.configuration.acceptedTestsLocation}/${testCaseXMLObject.classname}`,
-			time:time(testCaseXMLObject.time)
+			url: `${GITHUB_REPO_URL}${this.configuration.repo}/tree/${this.configuration.github.branch}/${testCaseXMLObject.classname}`,
+			location:testCaseXMLObject.classname,
+			time: time(testCaseXMLObject.time)
 		};
+
 		if (testCaseXMLObject.failure) {
 			testCase.status = 'ko';
 			testCase.message = testCaseXMLObject.failure.message;
 		}
 		return testCase;
 	}
 
-	parse(xUnitFilePath, callback) {
+	parse(xUnitFilePath) {
 		const self = this;
-		this.readFile(xUnitFilePath, (err, data) => {
-			if (!err) {
-				xml2js.parseString(data, function (err, parsedData) {
-					if (err) {
-						callback(err);
-					} else {
-						if (parsedData.testsuite.$.tests !== '0') {
-							parsedData = from(parsedData);
-							parsedData.suite.timestamp = new Date(parsedData.suite.timestamp).getTime();
-							if (parsedData.suite && parsedData.suite.tests) {
-								const testCases = [];
-								parsedData.suite.tests.forEach((testCaseXMLObject) => {
-									var testCase = self.parseSingleTestData(testCaseXMLObject, parsedData);
-									testCases.push(testCase);
-								});
-								let testSuite = {
-									name: parsedData.suite.name,
-									tests: parsedData.suite.summary.tests,
-									failures: parsedData.suite.summary.failures,
-									timestamp: `${parsedData.suite.timestamp || self.now.getTime()}`,
-									testCases: testCases
-								};
-								if (!parsedData.suite.timestamp) {
-									console.log('Warning : No timestamp was provided for specified test suite, using current system date.');
+		return new Promise((resolve, reject) => {
+			this.readFile(xUnitFilePath, (err, data) => {
```

Comment:
If you switch to a Promise API, why not also switch `readFile` to return a Promise?